test/unit/libraries/ReservoirLibrary.t.sol

```diff
@@ -0,0 +1,239 @@
+pragma solidity ^0.8.0;
+
+import "v3-core/test/__fixtures/BaseTest.sol";
+
+import { ExtraData } from "src/interfaces/IReservoirRouter.sol";
+
+import { ReservoirLibrary } from "src/libraries/ReservoirLibrary.sol";
+
+contract ReservoirLibraryTest is BaseTest
+{
+    function testGetAmountOut_ErrorChecking(uint256 aAmountIn) public
+    {
+        // assume
+        uint256 lAmountIn = bound(aAmountIn, 1, type(uint112).max);
+
+        // act & revert
+        vm.expectRevert("RL: INSUFFICIENT_INPUT_AMOUNT");
+        ReservoirLibrary.getAmountOutStable(0, 10, 10, 30, ExtraData(0,0,0));
+
+        vm.expectRevert("RL: INSUFFICIENT_INPUT_AMOUNT");
+        ReservoirLibrary.getAmountOutConstantProduct(0, 10, 10, 30);
+
+        vm.expectRevert("RL: INSUFFICIENT_LIQUIDITY");
+        ReservoirLibrary.getAmountOutStable(lAmountIn, 0, 0, 0, ExtraData(0,0,0));
+
+        vm.expectRevert("RL: INSUFFICIENT_LIQUIDITY");
+        ReservoirLibrary.getAmountOutConstantProduct(lAmountIn, 0, 0, 0);
+    }
+
```

Comment:
Should split into `testGetAmountOut_InsufficientLiquidity` and `testGetAmountOut_InsufficientInputAmount`

Reply:
split for this as well as `testGetAmountIn`, fixed